aten/src/ATen/native/Convolution.cpp

```diff
@@ -787,8 +874,12 @@ at::Tensor conv1d(
   Tensor input;
   bool is_batched;
   std::tie(input, is_batched) = batchify(input_, /*num_spatial_dims=*/ 1, "conv1d");
-  auto output = at::_convolution_mode(
-      input, weight, bias, stride, std::move(padding), dilation, groups);
+  Tensor output;
+  if (at::isComplexType(input_.scalar_type())) {
```

Comment:
nice so adds support for complex half too

Reply:
Not at this point as `+`, `-`, `*`, `view_as_real` will need to be supported first.